### PR TITLE
[foundation] Xcode 13 & Xcode 14 beta 1-3 bindings

### PR DESCRIPTION
**Pull request by chamons**
This is a port of https://github.com/xamarin/xamarin-macios/pull/13328 to xcode 14, as that was closed and not landed due to the widespread attribute tsunami.

I had to disable a significant portion of the bindings due to https://github.com/xamarin/xamarin-macios/issues/15577 , which in short points out that some of the callbacks use generics, which requires a generator patch AND fails an ancient cecil test.

~~There are a handful of comments from Rolf on the previous PR related to disabled tests that I will apply tomorrow.~~ Fixed

However, I accidentally pushed Xcode 14 bindings into this PR, so it now includes Xcode 14 beta 1-3.

### CODE REVIEW

File: tests/monotouch-test/monotouch-test.sln [hunk 1]
```diff
@@ -0,0 +1,46 @@
+﻿
```
Comment (mandel-macaque) [inline tests/monotouch-test/monotouch-test.sln:1]: I don't think we need this file, do we?

**Review by mandel-macaque — APPROVED** [review]
Approving, although I might be approving my own code.

File: tests/monotouch-test/monotouch-test.sln (hunk 1 above)
Comment (chamons) [reply tests/monotouch-test/monotouch-test.sln]: Removed.

**Comment by chamons** [comment]
@dalexsoto @mandel-macaque - Please review these commits:

https://github.com/xamarin/xamarin-macios/pull/15579/commits/02f45557ff6c0d332b35bad58bf05077a4ef1e12
https://github.com/xamarin/xamarin-macios/pull/15579/commits/e9708806e8f66825e7420c5ef6356615e13a6fbc

as they are changed (or didn't make but commented on) I made based upon review comments Rolf made on the last PR.

**Review by dalexsoto — APPROVED** [review]

**Pull request merged**